packages/integration-sdk-cli/src/generator/template/src/config.ts.hbs

```diff
@@ -0,0 +1,57 @@
+import {
+  IntegrationExecutionContext,
+  {{#if configFields}}IntegrationValidationError,{{/if}}
+  IntegrationInstanceConfigFieldMap,
+  IntegrationInstanceConfig,
+} from '@jupiterone/integration-sdk-core';
+import { createAPIClient } from './client';
+
+/**
+ * A type describing the configuration fields required to execute the
+ * integration for a specific account in the data provider.
+ *
+ * When executing the integration in a development environment, these values may
+ * be provided in a `.env` file with environment variables. For example:
+ *
+ * - `CLIENT_ID=123` becomes `instance.config.clientId = '123'`
+ * - `CLIENT_SECRET=abc` becomes `instance.config.clientSecret = 'abc'`
+ *
+ * Environment variables are NOT used when the integration is executing in a
+ * managed environment. For example, in JupiterOne, users configure
+ * `instance.config` in a UI.
+ */
+export const instanceConfigFields: IntegrationInstanceConfigFieldMap = {
+  {{#each configFields}}
+  {{camelCase field}}: {
+    type: '{{type}}',
+    mask: {{mask}}
+  }
+  {{/each}}
+};
+
+/**
+ * Properties provided by the `IntegrationInstance.config`. This reflects the
+ * same properties defined by `instanceConfigFields`.
+ */
+export interface IntegrationConfig extends IntegrationInstanceConfig {
+  {{#each configFields}}
+  {{camelCase field}}: {{ type }}
+  {{/each}}
+}
+
+export async function validateInvocation(
+  context: IntegrationExecutionContext<IntegrationConfig>,
+) {
+  const { config } = context.instance;
+
+  {{#if configFields}}
+  if ({{#each configFields}}!config.{{camelCase field}}{{#unless @last}}||{{/unless}}{{/each}}) {
```

Comment:
Separately...

We should consider adding a utility to the SDK that allows for default config validation. We copy/paste this code in every repo pretty much. Only a few repos have special validation.